import-data.sh

```diff
@@ -3,40 +3,129 @@
 # the bastion is already setup with synapse, mongoimport and mongofiles
 # command line clients
 #!/bin/bash
-set -e
+set -euo pipefail  # -e (exit on error), -u (treat unset variables as an error), -o pipefail (fail if any command in a pipeline fails)
+
+# Validate required arguments
+if [ $# -ne 5 ]; then
+    echo "Usage: $0 <BRANCH> <SYNAPSE_PASSWORD> <DB_HOST> <DB_USER> <DB_PASS>"
+    exit 1
+fi
 
 BRANCH=$1
 SYNAPSE_PASSWORD=$2
 DB_HOST=$3
 DB_USER=$4
 DB_PASS=$5
 
-CURRENT_DIR=$(pwd)
-WORKING_DIR=$CURRENT_DIR
-DATA_DIR=$WORKING_DIR/data
+# Database configuration
+readonly DB_NAME="model-ad"
+
+# Create secure MongoDB connection string - connect to target database
+MONGO_URI="mongodb://${DB_USER}:${DB_PASS}@${DB_HOST}/${DB_NAME}?authSource=admin"
+
+# Setup logging
+log() {
+    echo -e "\033[32m[LOG][$(date '+%Y-%m-%d %H:%M:%S')] $*\033[0m"
+}
+
+warn() {
+    echo -e "\033[33m[WARN][$(date '+%Y-%m-%d %H:%M:%S')] $*\033[0m" >&2
+}
+
+error() {
+    echo -e "\033[31m[ERROR][$(date '+%Y-%m-%d %H:%M:%S')] $*\033[0m" >&2
+}
+
+# Setup paths and directories
+readonly SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+readonly WORKING_DIR="${SCRIPT_DIR}"
+readonly DATA_DIR="${WORKING_DIR}/data"
+
+# Define collections to import (order matters for dependencies)
```

Comment:
What dependencies? How would someone know where to add a new collection?

Reply:
Whoops!  That's not true anymore.  I'll update it in the next commit.